client/blb/client.go

```diff
@@ -1122,39 +1141,127 @@ func (cli *Client) readOneTractReplicated(
 			log.V(1).Infof("read %s from tsid %d: no host", tract.Tract, tract.TSIDs[n])
 			continue
 		}
-
-		var read int
-		read, err = cli.tractservers.ReadInto(ctx, host, tract.Tract, tract.Version, thisB, thisOffset)
-		if err == core.ErrVersionMismatch {
-			badVersionHost = host
-		}
-		if err != core.NoError && err != core.ErrEOF {
-			log.V(1).Infof("read %s from tractserver at address %s: %s", tract.Tract, host, err)
-			// If we failed to read from a TS, report that to the curator. Defer so we can examine
-			// result.err at the end, to see if we succeeded on another or not.
-			defer func(host string, err core.Error) {
-				couldRecover := result.err == core.NoError || result.err == core.ErrEOF
-				go cli.curators.ReportBadTS(context.Background(), curAddr, tract.Tract, host, "read", err, couldRecover)
-			}(host, err)
-			continue // try another host
-		}
-		log.V(1).Infof("read %s from tractserver at address %s: %s", tract.Tract, host, err)
-		// In case of short read, i.e., read < len(thisB), we should
-		// pad the result with 0's. This doesn't need to be done
-		// explicitly if 'thisB' given to us starts to be empty.
-		// However, just in case it's not, we will ensure that by
-		// overwritting untouched bytes with 0's.
-		for i := read; i < len(thisB); i++ {
-			thisB[i] = 0
+		skipHost, badVersionHost, err = cli.readIntoReportBadTS(ctx, curAddr, host, nil, "", result, tract, thisB, thisOffset)
+		if skipHost {
+			continue
 		}
-		*result = tractResult{len(thisB), read, err, badVersionHost}
-		return
+		return // if skipHost is false, we have hit a short read or there was no error returned.
 	}
-
 	log.V(1).Infof("read %s all hosts failed", tract.Tract)
 	*result = tractResult{0, 0, err, badVersionHost}
 }
 
+func (cli *Client) readIntoReportBadTS(
```

Comment:
There's an error and a badVersionHost in `result`. What's the difference between the return values and the values that get returned in `result`? We only need one or the other, right?

Reply:
that makes sense. Will pull the error from the result.